org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/ResourceSetBasedAllContainersState.java

```diff
@@ -9,36 +9,46 @@
 package org.eclipse.xtext.resource.containers;
 
 import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import org.eclipse.emf.common.util.URI;
 
-import com.google.common.base.Joiner;
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.Multimap;
-import com.google.common.collect.Multimaps;
-import com.google.common.collect.SetMultimap;
-import com.google.common.collect.Sets;
-
 /**
  * This implementation {@link IAllContainersState} associates resource (e.g. their URIs) to containers. It assumes that
- * all URIs and their containers are known when {@link #configure(List, Multimap)} is called.
+ * all URIs and their containers are known when {@link #configure(List, Map)} is called.
  * 
  * @see FlatResourceSetBasedAllContainersState
  * 
  * @author Sven Efftinge - Initial contribution and API
  */
 public class ResourceSetBasedAllContainersState implements IAllContainersState {
-	
-	private SetMultimap<String, URI> container2URIs;
-	private SetMultimap<URI, String> uri2container;
+
+	private Map<String, Set<URI>> container2URIs;
```

Comment:
Previous usages of container2URIs would never return null when queried via `get(someHandle)`. getContainedURIs might need refinedment.
Into the bargain, it used to be possible to add items to the collection returned from getContainedURis and the changed would be reflected in the containers state, regardless of the fact whether the containers was initially known or not.
We don't (ab)use that detail in the framework, but I'm not a fan of the subtle change nor of the effort we'd need to put into retained that behaviro without using Guava Multimaps.

Reply:
> Previous usages of container2URIs would never return null when queried via `get(someHandle)`. getContainedURIs might need refinedment.

I adjusted `getContainedURIs` to reflect that, i.e. to `return container2URIs.getOrDefault(containerHandle, Set.of());`.

As a side-note, for `uri2container.get(uri)` the result was still checked for null.

> Into the bargain, it used to be possible to add items to the collection returned from getContainedURis and the changed would be reflected in the containers state, regardless of the fact whether the containers was initially known or not. We don't (ab)use that detail in the framework, but I'm not a fan of the subtle change nor of the effort we'd need to put into retained that behaviro without using Guava Multimaps.

I would call this an undesired implementation detail on which callers shouldn't rely on. Furthermore the documentation of this class states 
https://github.com/eclipse/xtext/blob/9e3b1b496e157a58fe3bf556bb910f03c3a7a9a7/org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/ResourceSetBasedAllContainersState.java#L25-L26

According to this contract it would not be allowed to add elements. And as a third point, adding elements to a value collection of `container2URIs`, would mean that it goes out of sync with `uri2container`.